std/src/main/scala/algebra/std/boolean.scala

```diff
@@ -2,20 +2,40 @@ package algebra
 package std
 
 import algebra.lattice.Bool
+import algebra.ring.CommutativeRig
 
 package object boolean extends BooleanInstances
 
 trait BooleanInstances {
-  implicit val booleanAlgebra = new BooleanAlgebra
+  implicit val booleanAlgebra: BooleanAlgebra =
+    new BooleanAlgebra
 }
 
-class BooleanAlgebra extends Bool[Boolean] with Order[Boolean] {
+class BooleanAlgebra extends Bool[Boolean] with Order[Boolean] with CommutativeRig[Boolean] with Serializable {
```

Comment:
maybe a comment that this is different than the CommutativeRig that is obtained from the field GF(2) (which has plus == xor).

This is a case where the most constrained structure (Field being more constrained than Rig) may not be what you are looking for.